### PR TITLE
Add support for jag.disable and jag.timeout

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: cmd/jag/commands/device.go [hunk 1]
```diff
@@ -73,15 +73,15 @@ func (d Device) Ping(ctx context.Context, sdk *SDK) bool {
...
+func (d Device) Run(ctx context.Context, sdk *SDK, b []byte, defines string) error {
 	req, err := http.NewRequestWithContext(ctx, "PUT", d.Address+"/code", bytes.NewReader(b))
 	if err != nil {
 		return err
 	}
 	req.Header.Set(JaguarDeviceIDHeader, d.ID)
 	req.Header.Set(JaguarSDKVersionHeader, sdk.Version)
-	if len(options) > 0 {
-		req.Header.Set(JaguarRunOptionsHeader, options)
+	if len(defines) > 0 {
```
Comment (floitsch) [inline cmd/jag/commands/device.go:83]: Personally I prefer to check against "":
```suggestion
	if defines != "" {
```
Fine either way, though.

File: src/jaguar.toit [hunk 2]
```diff
@@ -26,7 +26,10 @@ RUN_DEFINES_HEADER ::= "X-Jaguar-Run-Defines"
 
 logger ::= log.Logger log.INFO_LEVEL log.DefaultTarget --name="jaguar"
 validate_firmware / bool := firmware.is_validation_pending
-disabled / monitor.Signal? := null
+
+disabled / bool := false
```
Comment (floitsch) [inline src/jaguar.toit:30]: Add a comment what this global does.
It looks like it's used as an indication that Jaguar should disable itself, and not that it is already disabled.
Maybe a better name would be `program_needs_full_network` or something like this.

File: src/jaguar.toit [hunk 3]
```diff
@@ -26,7 +26,10 @@ RUN_DEFINES_HEADER ::= "X-Jaguar-Run-Defines"
...
+network_down / monitor.Semaphore ::= monitor.Semaphore
```
Comment (floitsch) [inline src/jaguar.toit:31]: Would it make sense to call this `network_released`?

**Review by floitsch — APPROVED** [review]
Still LGTM.

File: src/jaguar.toit (hunk 3 above)
Comment (kasperl) [reply src/jaguar.toit:31]: Went with `network_free`.

File: src/jaguar.toit (hunk 2 above)
Comment (kasperl) [reply src/jaguar.toit:30]: Done.

File: cmd/jag/commands/device.go (hunk 1 above)
Comment (kasperl) [reply cmd/jag/commands/device.go]: Done.

**Pull request merged**